### PR TITLE
process.util: improve modularity of get_citekeys_df and generate_csl_items

### PR DESCRIPTION
**Pull request by dhimmel**
I've resumed work on #99, which needs to access some functions from manubot.cite.util in a more general purpose way (without requiring argparse args for example).

These commits are currently in that PR, but may make sense to merge first, since #99 could take a while.

### CODE REVIEW

File: manubot/process/util.py [hunk 1]
```diff
@@ -285,51 +307,50 @@ def get_citekeys_df(args, text):
...
+        citekey_aliases = dict(
+            zip(tag_df["manuscript_citekey"], tag_df["detagged_citekey"])
         )
     else:
-        citekeys_df["detagged_citekey"] = None
+        citekey_aliases = {}
         logging.info(
             f"missing {args.citation_tags_path} file: no citation tags (citekey aliases) set"
         )
-    citekeys_df.detagged_citekey.fillna(
```
Comment (agitter) [inline manubot/process/util.py]: Does removing this `fillna` change the final data frame?

File: manubot/process/util.py [hunk 2]
```diff
@@ -285,51 +307,50 @@ def get_citekeys_df(args, text):
...
-    requests  # require `import requests` in case this is essential for monkey patching by requests_cache.
-    requests_cache.install_cache(args.requests_cache_path, include_get_headers=True)
-    cache = requests_cache.get_cache()
-    if args.clear_requests_cache:
-        logging.info("Clearing requests-cache")
-        requests_cache.clear()
-    logging.info(
-        f"requests-cache starting with {len(cache.responses)} cached responses"
-    )
+    Parameters
```
Comment (agitter) [inline manubot/process/util.py]: Could list the other two parameters here: `requests_cache_path` and `clear_requests_cache`

File: manubot/process/util.py (hunk 1 above)
Comment (dhimmel) [reply manubot/process/util.py]: No the functionality has been moved to:

```python
    citekeys_df["detagged_citekey"] = citekeys_df.manuscript_citekey.map(
        lambda citekey: citekey_aliases.get(citekey, citekey)
    )
```

File: manubot/process/util.py (hunk 2 above)
Comment (dhimmel) [reply manubot/process/util.py]: done in https://github.com/manubot/manubot/pull/190/commits/0d1c6d113f23db142b8dcb8af805613cc212e1a1

**Comment by dhimmel** [comment]
> My understanding is that there aren't any behavior changes with this refactoring. Is that correct?

Yes there should be no behavior changes.

**Comment by agitter** [comment]
Thanks for the clarifications and docstring updates.

**Pull request merged**